ir/visitor.cpp

```diff
@@ -130,7 +135,7 @@ class Visitor::ChangeTracker {
     void revisit_visited() {
         for (auto it = visited.begin(); it != visited.end();) {
             if (!it->second.visit_in_progress)
-                it = visited.erase(it);
+                visited.erase(it++);
```

Comment:
I had to look up why you make this change: I see that Abseil returns void on erase instead of an iterator.

Reply:
Yes. Unfortunately it's something they are not compatible with std maps. Though it is explicitly specified that post-incrementing iterator would work.

Reply:
Please add a comment about this (the post-increment). This is non-intuitive for anyone not very familiar with abseil maps.

Reply:
@vlstill I also thought about asking for a comment but I'm in two minds about it:
* It is different to the STL interface.
* If the Abseil collections spread to more places, are we going to add a comment every time the Abseil interface differs from the STL interface?

I'm happy to accept the argument that Visitor requires extra documentation because it is a core class, and so more people will look at the code to better understand how the compiler works/improve its performance.

Reply:
Let's add the comment for now. I believe this is the only place where abseil maps are different interface-wise (sans heterogeneous lookup, but we will have it for `unordered_map` with C++20).

The behavior wrt pointer / iterators stability is different, but I will take care about documentation separately